addons/website_blog/controllers/main.py

```diff
@@ -136,13 +138,28 @@ def blog(self, blog=None, tag=None, page=1, **opt):
         if date_begin and date_end:
             domain += [("create_date", ">=", date_begin), ("create_date", "<=", date_end)]
 
-        blog_url = QueryURL('', ['blog', 'tag'], blog=blog, tag=tag, date_begin=date_begin, date_end=date_end)
+        if request.env.user.has_group('base.group_website_designer'):
+            for group in request.env['blog.post'].read_group(domain, ['website_published'], groupby=["website_published"]):
+                if group['website_published']:
+                    published_count = group['website_published_count']
+                else:
+                    unpublished_count = group['website_published_count']
+            if state == "published":
+                domain += [("website_published", "=", True)]
+            elif state == "unpublished":
+                domain += [("website_published", "=", False)]
 
-        blog_post_ids = blog_post_obj.search(cr, uid, domain, order="create_date desc", context=context)
-        blog_posts = blog_post_obj.browse(cr, uid, blog_post_ids, context=context)
+        blog_url = QueryURL('', ['blog', 'tag'], blog=blog, tag=tag, state=state, date_begin=date_begin, date_end=date_end)
```

Comment:
Why do you write a new url by hand instead of using a QueryURL that will do it for you ?